nsq/deflate_socket.py

```diff
@@ -38,8 +38,14 @@ def _recv(self, size, method):
         return uncompressed
 
     def send(self, data):
-        if isinstance(data, memoryview):
-            data = data.tobytes()
-        chunk = self._compressor.compress(data)
-        self._socket.send(chunk + self._compressor.flush(zlib.Z_SYNC_FLUSH))
-        return len(data)
+        return self._socket.send(data)
+
+
+class DeflateEncoder(object):
+
+    def __init__(self, level):
+        self._compressor = zlib.compressobj(level, zlib.DEFLATED, -zlib.MAX_WBITS)
```

Comment:
we no longer need lines 10 and 12 in this file as a result of this moving here

Reply:
Done.